annotation-file-utilities/src/annotator/find/ASTPathCriterion.java

```diff
@@ -383,9 +383,8 @@ private Tree getNext(Tree actualNode, ASTPath astPath, int ix) {
                 } else if (astNode.childSelectorIs(ASTPath.BOUND)) {
                   return arg < 0 ? clazz.getExtendsClause()
                           : clazz.getImplementsClause().get(arg);
-                } else {
-                    return null;
                 }
+                return null;
```

Comment:
That's weird.  Every clause in the if statement that starts at line 374 ends with a return statement, so why does the compiler think there can be a fallthrough?
How about putting
```
    throw new Exception("This can't happen"); // suppress spurious javac fallthrough warning
```
on line 387, and retain the `return null;` between lines 382 and 383?